.github/workflows/check.yml

```diff
@@ -1,71 +1,66 @@
-name: "Check CML GitHub Action"
-
+name: Check CML GitHub Action
 on: [push]
-
 jobs:
   check:
-    if: "!contains(github.event.head_commit.message, '[ci skip]')"
+    defaults:
+      run:
+        shell: bash
     strategy:
       matrix:
-        runs-on: [ '{"runs-on": "ubuntu-latest"}', 
-          '{"runs-on": "macos-latest"}', 
-          '{"runs-on": "ubuntu-latest", "container": "ubuntu:18.04"}', 
-          '{"runs-on": "windows-latest"}']
-
-    runs-on: ${{ fromJSON(matrix.runs-on).runs-on }}
-    container: ${{ fromJSON(matrix.runs-on).container }} 
-
+        system:
+          - ubuntu-latest
+          - macos-latest
+          - windows-latest
+        include:
+          - system: ubuntu-18.04
+            container: ubuntu:18.04
+    runs-on: ${{ matrix.system }}
+    container: ${{ matrix.container }}
     steps:
-      - name: "git"
-        if: "fromJSON(matrix.runs-on).container == 'ubuntu:18.04'"
+      - name: install deps
```

Comment:
Just nipticking: have we standardiced on casing and abbreviations for step names? I personally like them in sentence case and without abbreviations, like in the GitHub-generated ones, but I wouldn't mind them being otherwise as long as we always use the same convention.